### PR TITLE
Feature/selectpcs

### PR DESCRIPTION
**Pull request by SamGoodeySFW**

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Scheme/Controllers/HomeController.cs [hunk 1]
```diff
@@ -185,9 +185,11 @@ public async Task<ActionResult> ChooseActivity(ChooseActivityViewModel viewModel
 
                         var aatfReturnId = await client.SendAsync(User.GetAccessToken(),
                             new AddReturn() { OrganisationId = viewModel.OrganisationId });
-
+                        /*
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Scheme/Controllers/HomeController.cs]: Remove the commented out code

File: src/EA.Weee.Core/Validation/MinimumElementAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿namespace EA.Weee.Core.Validation
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.ComponentModel.DataAnnotations;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Core/Validation/MinimumElementAttribute.cs]: Can we change the attribute so that it takes a string that is the error message to display then we can re-use easily elsewhere

File: src/EA.Weee.Web.Tests.Unit/Areas/AatfReturn/ViewModels/SelectYourPCSViewModelTests.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        [Fact]
+        public void SelectYourPCSViewModel_GivenSelectedSchemesListIsNull_ArgumentNullExceptionExpected()
+        {
+            Action action = () =>
+            {
+                var selectYourPCSModel = new SelectYourPCSViewModel(A.Fake<List<SchemeData>>(), null);
+            };
+
+            action.Should().Throw<ArgumentNullException>();
+        }
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/Areas/AatfReturn/ViewModels/SelectYourPCSViewModelTests.cs]: Add a test to ensure that the SelectedScheme property has the Minimum attribute, have a look at the NonObligatedCategoryValue tests as example.

File: src/EA.Weee.Domain/AatfReturn/ReturnScheme.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    using EA.Prsd.Core;
+    using EA.Prsd.Core.Domain;
+
+    public partial class ReturnScheme : Entity
+    {
+        protected ReturnScheme()
+        {
+        }
+
+        public ReturnScheme(Scheme scheme, Return @return)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Domain/AatfReturn/ReturnScheme.cs]: You can add some unit tests to ensure that the ReturnScheme catches the null exception when the parameters are supplied as null

File: src/EA.Weee.RequestHandlers/AatfReturn/AddReturnSchemeHandler.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            authorization.EnsureCanAccessExternalArea();
+
+            var scheme = await schemeDataAccess.GetSchemeOrDefault(message.SchemeId);
+
+            var @return = await returnDataAccess.GetById(message.ReturnId);
+            var returnScheme = new ReturnScheme(scheme, @return);
+
+            await returnSchemeDataAccess.Submit(returnScheme);
+
+            return returnScheme.Id;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/AatfReturn/AddReturnSchemeHandler.cs]: Add a test to check the return value

File: src/EA.Weee.Core/Validation/EnsureOneElementAttribute.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+﻿namespace EA.Weee.Core.Validation
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.ComponentModel.DataAnnotations;
+    using System.Linq;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Core/Validation/EnsureOneElementAttribute.cs]: Do we still need this one or can it be removed?

File: src/EA.Weee.RequestHandlers/AatfReturn/GetSchemeRequestHandler.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+                    {
+                        pcsListData.SchemeId = item.Id;
+                        pcsListData.PCSName = item.SchemeName;
+                        pcsListData.ApprovalNumber = item.ApprovalNumber;
+                    }
+                }
+            }
+            return pcsListData;
+        }
+    }
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/AatfReturn/GetSchemeRequestHandler.cs]: Do this via a Scheme list to PCSdata mapper and related unit tests

File: src/EA.Weee.Web/Areas/AATFReturn/Controllers/SelectYourPCSController.cs [hunk 8]
```diff
@@ -31,19 +31,54 @@ public SelectYourPCSController(Func<IWeeeClient> apiclient, BreadcrumbService br
...
+                    }
+                }
+                return RedirectToAction("Index", "AatfTaskList",
+                                            new { area = "AatfReturn", organisationId = viewModel.OrganisationId, returnId = viewModel.ReturnId });
+            }
+            else
+            {
+                using (var client = apiClient())
+                {
+                    viewModel.SchemeList = await client.SendAsync(User.GetAccessToken(), new GetSchemesExternal());
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/AATFReturn/Controllers/SelectYourPCSController.cs]: We shouldnt need to re-retrieve the schemes  list if we have set the view and view model correct as they should get bound by the model binder... 

It should just be a return View(viewModel) when the ModelState isnt valid

File: src/EA.Weee.Web/Areas/AATFReturn/Controllers/SelectYourPCSController.cs [hunk 9]
```diff
@@ -31,19 +31,54 @@ public SelectYourPCSController(Func<IWeeeClient> apiclient, BreadcrumbService br
...
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public virtual async Task<ActionResult> Index(SelectYourPCSViewModel viewModel)
+        {
+            if (ModelState.IsValid)
+            {
+                using (var client = apiClient())
+                {
+                    foreach (var scheme in viewModel.SelectedSchemes)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/AATFReturn/Controllers/SelectYourPCSController.cs]: Implement a request creator to create the request rather than doing this in the action... and related unit tests :)

File: src/EA.Weee.RequestHandlers/Scheme/GetSchemesExternalHandler.cs [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    using Requests.Scheme;
+
+    public class GetSchemesExternalHandler : IRequestHandler<GetSchemesExternal, List<SchemeData>>
+    {
+        private readonly IGetSchemesDataAccess dataAccess;
+        private readonly IMap<Scheme, SchemeData> schemeMap;
+        private readonly IWeeeAuthorization authorization;
+
+        public GetSchemesExternalHandler(
+    IGetSchemesDataAccess dataAccess,
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Scheme/GetSchemesExternalHandler.cs]: Indent these :)

File: src/EA.Weee.Web/Areas/AATFReturn/Controllers/ReceivedPCSListController.cs [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        // GET: AatfReturn/ReceivedPCSList
+        [HttpGet]
+        public async Task<ActionResult> Index(Guid organisationId, Guid returnId)
+        {
+            /*
+            var viewModel = new ReturnViewModel();
+            using (var client = apiClient())
+            {
+                var @return = await client.SendAsync(User.GetAccessToken(), new GetReturnScheme(returnId));
+                viewModel = mapper.Map<ReturnViewModel>(@return);
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/AATFReturn/Controllers/ReceivedPCSListController.cs]: Remove this commented out code
